<img src="images/usm.jpg" width="480" height="240" align="left"/>

# MAT281 - Laboratorio N°03

## Objetivos de la clase

* Reforzar los conceptos básicos de pandas.

## Contenidos

* [Problema 01](#p1)


## Problema 01


<img src="https://imagenes.universia.net/gc/net/images/practicas-empleo/p/pr/pro/profesiones-con-el-avance-de-la-tecnologia.jpg" width="480" height="360" align="center"/>


EL conjunto de datos se denomina `ocupation.csv`, el cual contiene información tal como: edad ,sexo, profesión, etc.

Lo primero es cargar el conjunto de datos y ver las primeras filas que lo componen:

In [1]:
import pandas as pd
import os

In [35]:
# cargar datos
df = pd.read_csv(os.path.join("data","ocupation.csv"), sep="|").set_index('user_id')
df.head()

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

In [36]:
#limpiamos los datos
df1=df[df.notnull().all(axis=1)]
print(df1.shape)
type(df1['zip_code'][1])

(943, 4)


str

Se determina que el conjunto de datos no tenía valores nulos.

El objetivo es tratar de obtener la mayor información posible de este conjunto de datos. Para cumplir este objetivo debe resolver las siguientes problemáticas:

1. ¿Cuál es el número de observaciones en el conjunto de datos?

In [37]:
df.shape

(943, 4)

Se observa que el número de observaciones del conjunto de datos corresponde a 943 filas.

2. ¿Cuál es el número de columnas en el conjunto de datos?

Del formato del dataframe vemos que tenemos 4 columnas.

3. Imprime el nombre de todas las columnas

In [38]:
df.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

4. Imprima el índice del dataframe

In [39]:
df.index

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

5. ¿Cuál es el tipo de datos de cada columna?

In [40]:
df.dtypes

age            int64
gender        object
occupation    object
zip_code      object
dtype: object

Observamos que:
- age es de tipo entero
- gender y occupation son de tipo objeto
- zip_code es de tipo objeto por tanto los números en zip_code están definidos como string.

6. Resumir el conjunto de datos

In [41]:
df.describe()

age
count  943.000000
mean    34.051962
std     12.192740
min      7.000000
25%     25.000000
50%     31.000000
75%     43.000000
max     73.000000

7. Resume conjunto de datos  con todas las columnas

In [42]:
df.describe(include='all')

age gender occupation zip_code
count   943.000000    943        943      943
unique         NaN      2         21      795
top            NaN      M    student    55414
freq           NaN    670        196        9
mean     34.051962    NaN        NaN      NaN
std      12.192740    NaN        NaN      NaN
min       7.000000    NaN        NaN      NaN
25%      25.000000    NaN        NaN      NaN
50%      31.000000    NaN        NaN      NaN
75%      43.000000    NaN        NaN      NaN
max      73.000000    NaN        NaN      NaN

8. Imprimir solo la columna de **occupation**.

In [43]:
df['occupation']

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

9. ¿Cuántas ocupaciones diferentes hay en este conjunto de datos?

In [44]:
len(df['occupation'].unique())

21

Se determina que hay 21 ocupaciones distintas, donde una categoría es 'otros'. En resumen hay 20 ocupaciones especificas distintas y una categoría 'otros' donde se agrupan todas las ocupaciones distintas a las anteriores 20.

10. ¿Cuál es la ocupación más frecuente?

De la descripción de los datos se determina que la ocupación más frecuente es la de 'estudiante' con 196 ocurrencias en el conjunto de datos.

11. ¿Cuál es la edad media de los usuarios?

De la descripcion de la columna edad se determina que la edad media es de 34 años

12. ¿Cuál es la edad con menos ocurrencia?

In [56]:
lista_edades=df['age'].unique()    #SACAMOS LAS EDADES DISTINTAS DEL CONJUNTO DE DATOS
df_age=df['age']                   #DEFINIMOS LA COLUMNA DE EDADES DEL CONJUNTO DE DATO
edad_menos_frecuente=0             #INICIALIZAMOS LA EDAD MENOS FRECUENTE COMO 0
frecuencia=1000                    #DEFINIMOS UNA FRECUENCIA QUE SUPERA EL NÚMERO DE DATOS PARA QUE CAMBIE CON EL PRIMER
                                   #CHEQUEO
for i in lista_edades:
    frec=len(df_age[df_age==i])    #PARA CADA EDAD DISTINTA CALCULAMOS EL NÚMERO DE OCURRENCIAS EN EL CONJUNTO DE DATOS
    if frec<frecuencia:            #SE GUARDA EL ELEMENTO CON MENOR FRECUENCIA
        frecuencia=frec
        edad_menos_frecuente=i
        
edad_menos_frecuente
print("la frecuencia es:", frecuencia)
print('la edad menos frecuente es:',edad_menos_frecuente)

la frecuencia es: 1
la edad menos frecuente es: 7


Del análisis anterior se desprende que la edad menos frecuente es la de 7 años y está tiene solo una ocurrencia el
conjunto de datos.